src/debugger/nodeDebugWrapper.ts

```diff
@@ -1,194 +1,246 @@
+                                })
+                                .then(() =>
+                                    nodeDebugWrapper.attachRequest(originalRequest, this, request, args, mobilePlatform));
+                        }
+                    }).catch(error =>
+                        nodeDebugWrapper.bailOut(this, error.message));
             });
+        };
+    }
 
-            // We do not permit arbitrary args to be passed to our process
-            args.args = [
-                args.platform,
-                debugServerListeningPort.toString(),
-                !isNullOrUndefined(args.iosRelativeProjectPath) ? args.iosRelativeProjectPath : IOSPlatform.DEFAULT_IOS_PROJECT_RELATIVE_PATH,
-                args.target || "simulator",
-            ];
+    /**
+     * Intecept the "attachRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeAttachRequest(): void {
+        const nodeDebugWrapper = this;
+        this.nodeDebugSession.prototype.attachRequest = function (request: any, args: IAttachRequestArgs) {
+            nodeDebugWrapper.requestSetup(this, args);
+            nodeDebugWrapper.attachRequest(nodeDebugWrapper.nodeDebugSession.prototype.launchRequest, this, request, args, null);
+        };
+    }
 
-            if (!isNullOrUndefined(args.logCatArguments)) { // We add the parameter if it's defined (adapter crashes otherwise)
-                args.args = args.args.concat([parseLogCatArguments(args.logCatArguments)]);
-            }
+    /**
+     * Intecept the "disconnectRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeDisconnectRequest(): void {
+        const originalRequest = this.nodeDebugSession.prototype.disconnectRequest;
+        const nodeDebugWrapper = this;
+
+        this.nodeDebugSession.prototype.disconnectRequest = function (response: any, args: any): void {
+            // First we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
+            nodeDebugWrapper.remoteExtension.stopMonitoringLogcat()
+                .catch(reason =>
+                    Log.logError(`WARNING: Couldn't stop monitoring logcat: ${reason.message || reason}\n`))
+                .finally(() =>
+                    originalRequest.call(this, response, args));
+        };
+    }
 
-            originalNodeDebugSessionLaunchRequest.call(this, request, args);
+    /**
+     * Makes the required setup for request customization
+     * - Enables telemetry
+     * - Sets up mobilePlatformOptions, remote extension and projectRootPath
+     * - Starts debug server
+     * - Create global logger
+     */
+    private requestSetup(debugSession: NodeDebugSession, args: any) {
+        this.projectRootPath = path.resolve(args.program, "../..");
+        this.remoteExtension = RemoteExtension.atProjectRootPath(this.projectRootPath);

```

Comment:
Nevermind, I see. It's for the pipeline to communicate with the extension process if I am correct.